pkgs/development/tools/yarn2nix-moretea/yarn2nix/lib/generateNix.js

```diff
@@ -103,7 +104,14 @@ function fetchLockedDep(builtinFetchGit) {
       return fetchgit(fileName, urlForGit, rev, branch || 'master', builtinFetchGit)
     }
 
-    const [algo, hash] = integrity ? integrity.split('-') : ['sha1', sha1OrRev]
+    // Pull out integrity hash, providing a default and using the "best" algorithm if there are multiple.
+    let algo = "sha1";
+    let hash = sha1OrRev;
+    if (integrity) {
+      const integrities = ssri.parse(integrity);
```

Comment:
This looks good but apparently, it might choose unsupported hash type. Nix [supports](https://nixos.org/manual/nix/stable/language/advanced-attributes.html#adv-attr-outputHashAlgo) `sha1`, `sha256` and `sha512`, while [`ssri.parse`](https://github.com/npm/ssri#parse) can return ¿anything? (or `sha256`, `sha384`, and `sha512`) in strict mode).

Reply:
Thanks, I wasn't aware of that constraint! I almost threw out the `ssri` dependency to use a simple filter & sort method after learning that, but decided to leave it. This way we can handle parsing less-common stuff like integrity options, and it will be more obvious how to pick the best algorithm if nix supports new ones in the fewer.

Reply:
It does not seem this is fixed, is it?

Reply:
Nevermind, was reading commit per commit